proto/workflow.proto

```diff
@@ -176,6 +176,30 @@ message ExecuteWorkflowRequest {
 
   // If true, start the workflow but do not wait for the status to be returned.
   bool async = 14;
+
+  // Optional: Custom actions to execute. Correspond to actions in
+  // buildbuddy.yaml. If this is set, action_names will be ignored.
+  repeated CustomAction custom_action = 15;
+}
+
+message CustomAction {
```

Comment:
What if we just moved the definition of BuildBuddyConfig.Action to proto and refactored the code to use that everywhere? That way we can guarantee the two structs don't diverge

Reply:
Im ok with that as well. 😁

Reply:
Another option is we could use Zoey's yaml variable interpolation logic ([Example](https://github.com/buildbuddy-io/buildbuddy-internal/blob/master/enterprise/deployment/dev/us-west1-b/executor.dev.yaml#L10)) and just have users pass in a map of variable name -> value to the API
Ex.  variables: {
     "service-name": "login",
    ...
}

The advantage for that is if there's a lot of redundant information for each action (like image, env variables, resource requests), users won't have to pass in all that data

Reply:
I forgot that @vadimberezniker actually had added a `Runner` API a while back to support remote bazel, and I suspect this has some overlap. Can you check the runner API and see whether it would work instead of custom actions?

Reply:
(I think adding an `inputs` field to our existing workflows and allowing people to specify a mapping of inputs also seems reasonable, if the runner API needs too much work)

Reply:
I'd prefer exposing the runner API publicly to allowing custom workflow actions.